packages/keypair/src/generate-vanity-key-pair.ts

```diff
@@ -0,0 +1,49 @@
+import { createKeyPairSignerFromPrivateKeyBytes, type KeyPairSigner } from '@solana/kit'
+
+export interface GenerateVanityKeyPairProps {
+  caseSensitive?: boolean
+  prefix?: string
+  suffix?: string
+}
+
+export async function generateVanityKeyPair({
+  caseSensitive = true,
+  prefix = '',
+  suffix = '',
+}: GenerateVanityKeyPairProps): Promise<KeyPairSigner> {
+  if (!prefix && !suffix) {
```

Comment:
I think instead of generating a random key pair, this function should validate if either `prefix` or `suffix` have a length and if not, throw an error.